### ISSUE TITLE
Consider using a new CSS grammar: Support highlighting comments for JavaScript and CSS codes in HTML

### ISSUE THREAD

**Issue by bridgetowisdom** [microsoft/vscode#111528]
Issue Type: <b>Bug</b>

Hello,
While using Visual Studio Code, I noticed that attempting to comment JavaScript and CSS codes (for the sake of compatibility with very old browsers) produce highlighting errors in comments, a syntax error in JavaScript saying that an expression was expected when commenting the JavaScript, and a highlighting error in the rest of the code.

I am using Monokai as the colour theme for Visual Studio Code.
If you want an example to understand with it, please copy-paste the code below:
```html
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <!-- There is no precise example, it is just a case in general -->
    <style type="text/css">
        <!--
            element::after {color: blue;}
            element::before {color: white;}
        -->
        </style>
    <script language="JavaScript" type="text/javascript">
    <!--
        function checkWhat() {
            let check = true; // random things to type; still bad highlighting
        }
    //-->
    </script>
    <title>Variable scope testing</title>
</head>

<body>
    <p style="font-style: italic">Test</p>
    <noscript>JavaScript is disabled. If you are using Internet Explorer
    and it said that it blocked ActiveX contents, please allow them.</noscript>
</body>

</html>

```

As you see, commenting CSS code does not affect the CSS code itself, but still has it on the following HTML code.
Commenting the JavaScript code does not confuse the HTML code afterwards, but it shows two errors.

I hope you take supporting CSS and JavaScript codes into consideration.

Thank you in advance.

VS Code version: Code 1.51.1 (e5a624b788d92b8d34d1392e4c4d9789406efe8f, 2020-11-10T23:34:32.027Z)
OS version: Windows_NT x64 10.0.19042

<details><summary>Extensions (66)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-eclipse-keybindings|alp|0.12.0
vscode-intelephense-client|bme|1.5.4
vscode-swf-debug|bow|1.2.2
doxdocgen|csc|1.1.0
vscode-markdownlint|Dav|0.37.2
vscode-eslint|dba|2.1.13
xml|Dot|2.5.1
vscode-npm-script|eg2|0.3.13
prettier-vscode|esb|5.8.0
php-debug|fel|1.13.0
php-intellisense|fel|2.3.14
php-pack|fel|1.0.2
vscode-firefox-debug|fir|2.9.1
vscode-pull-request-github|Git|0.21.2
beautify|Hoo|1.5.0
discord-vscode|icr|4.0.0
better-cpp-syntax|jef|1.15.5
vscode-peacock|joh|3.8.0
classic-asp-html|jtj|0.1.2
intellij-idea-keybindings|k--|0.2.41
chat|kar|0.35.0
kite|kit|0.135.0
mp-rest-client-generator-vscode-ext|Mic|0.1.2
mp-starter-vscode-ext|Mic|0.2.5
vscode-microprofile-pack|Mic|0.1.2
vscode-docker|ms-|1.8.1
csharp|ms-|1.23.6
vscode-kubernetes-tools|ms-|1.2.1
mssql|ms-|1.9.0
python|ms-|2020.11.371526539
vscode-pylance|ms-|2020.11.2
jupyter|ms-|2020.11.372831992
remote-containers|ms-|0.148.1
remote-ssh|ms-|0.56.0
remote-ssh-edit|ms-|0.56.0
remote-wsl|ms-|0.51.4
cpptools|ms-|1.1.2
cpptools-extension-pack|ms-|1.0.0
powershell|ms-|2020.6.0
vsliveshare|ms-|1.0.3206
vsliveshare-audio|ms-|0.1.91
vsliveshare-pack|ms-|0.4.0
vsonline|ms-|1.0.3075
debugger-for-chrome|msj|4.12.11
vscode-react-native|msj|1.1.1
oracledevtools|Ora|19.3.3
vscode-boot-dev-pack|Piv|0.0.8
vscode-concourse|Piv|1.23.0
vscode-manifest-yaml|Piv|1.23.0
vscode-spring-boot|Piv|1.23.0
java|red|0.71.0
vscode-community-server-connector|red|0.23.9
vscode-microprofile|red|0.1.1
vscode-rsp-ui|red|0.23.5
vscode-yaml|red|0.13.0
vscode-autohotkey|sle|0.2.2
sonarlint-vscode|Son|1.18.0
vscodeintellicode|Vis|1.2.10
vscode-java-debug|vsc|0.29.0
vscode-java-dependency|vsc|0.15.0
vscode-java-pack|vsc|0.11.0
vscode-java-test|vsc|0.26.0
vscode-maven|vsc|0.26.0
vscode-spring-boot-dashboard|vsc|0.1.10
vscode-spring-initializr|vsc|0.6.1
vscode-icons|vsc|11.0.0

(1 theme extensions excluded)

</details>

![bug_screenshot_to_attach](https://user-images.githubusercontent.com/62489255/100641558-e863bb00-3337-11eb-96a9-9f3646cab625.png)

**Comment by aeschli** [comment]
I can see that in the lower right corner, the language is `html` (all lower case). The built-in HTML extension uses `HTML`.
Can you run with all extensions disabled (Command: `Developer: Run with Extensions disabled` and check whether you still see the issue?

**Comment by bridgetowisdom** [comment]
Thank you for your reply.
I found `Developer: Reload with extensions disabled` instead of what you specified; I hope they work the same.
Here is the highlighting with the extensions disabled:

![extensions_disabled](https://user-images.githubusercontent.com/62489255/100859355-6c7c8680-348f-11eb-8f5f-a34717b96e0b.png)

The issue is still present.

**Comment by alexr00** [comment]
It looks like this is an issue with our CSS grammar. We get our CSS grammar from https://github.com/octref/language-css. However, that repository doesn't accept issues, and it hasn't had an update in a while (@octref FYI). 

To fix this issue, we would need to adopt a new CSS grammar.

**Comment by bridgetowisdom** [comment]
Thank you for your kind reply and for your consideration in this issue.

I understand. Besides the issue with the CSS grammar; if this issue will be accepted to the backlog, I also hope there will be consideration in the JavaScript code as well, as it mentions an error when commenting it out: “Expression expected.” You can see the red line under the opening comment tag in HTML in the second picture that I have posted.

![image](https://user-images.githubusercontent.com/62489255/101390458-25452a00-38c3-11eb-8fee-d5972935ee73.png)

To show you more of the issue, here is the picture above, the third picture. You could see that I have commented out the style tag so that I can show the rest of the code. I have intentionally declared a new variable in the JavaScript section because I figured out that it also puts an error: “';' expected.”

In short, the problem is related to both CSS and JavaScript. It satisfies to hear that this problem could be considered, but I would appreciate it if you consider JavaScript as well.

Nevertheless, I highly thank you for investigating the problem and I am looking forward to seeing the improvement that could be added to Visual Studio Code.

(Please excuse my English. I am not a native speaker.)

**Comment by bridgetowisdom** [comment]
It seems that the issue did not gain popularity. Nevertheless, thank you for the kind consideration to this issue, hoping that there will be a consideration to fix it in future releases, wishing the best of luck.
